pallets/collator-selection/src/lib.rs

```diff
@@ -555,6 +578,126 @@ pub mod pallet {
 			Self::deposit_event(Event::InvulnerableRemoved { account_id: who });
 			Ok(())
 		}
+
+		/// Todo
+		///
+		/// Todo
+		#[pallet::call_index(7)]
+		#[pallet::weight(T::WeightInfo::increase_bond(T::MaxCandidates::get()))]
+		pub fn increase_bond(origin: OriginFor<T>, bond: BalanceOf<T>) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			let new_bond =
+				<Candidates<T>>::try_mutate(|candidates| -> Result<BalanceOf<T>, DispatchError> {
```

Comment:
Maybe a map keyed by account Id would suit better the candidates use case to avoid this loop.

Reply:
Done.